thinc/layers/concatenate.py

```diff
@@ -62,10 +62,12 @@ def backprop(d_output: Array2d) -> InT:
         start = widths[0]
         for bwd, width in zip(callbacks[1:], widths[1:]):
             dY = model.ops.as_contig(d_output[:, start : start + width])
-            if isinstance(dX, Ragged):
-                dX.data += bwd(dY).data
-            else:
-                dX += bwd(dY)
+            gradient = bwd(dY)
+            if gradient is not None:
+                if isinstance(dX, Ragged):
+                    dX.data += gradient.data
+                else:
+                    dX += gradient
```

Comment:
How about something like:

```suggestion
            gradient = bwd(dY)
            if add_gradients:
                dX += gradient
            elif add_gradients_data_attr:
                dX.data += gradient.data
```

And before the loop we do:

```
add_gradients = hasattr(dX, "__add__") or hasattr(dX, "__iadd__")
add_gradients_data = hasattr(dX, "data") and (hasattr(dX.data, "__add__") or hasattr(dX.data, "__iadd__"))
```

Reply:
That won't solve the problem of concatenating `None` though. The problem is that one of the callbacks (the `Tok2VecListener`) doesn't return a gradient if the listener is used multiple times in the pipeline.